### PR TITLE
1) test cases for task repo

### PR DESCRIPTION
**Pull request by farhaninfyom**

### CODE REVIEW

File: tests/Repositories/TaskRepositoryTest.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    /** @test */
+    public function it_can_store_task_with_tags_and_its_assigned_user()
+    {
+        $tag = factory(Tag::class)->create();
+        $assignees = factory(User::class)->times(2)->create();
+        $task = [
+            'title'       => $this->faker->title,
+            'description' => $this->faker->text,
+            'project_id'  => $this->faker->randomDigitNotNull,
```
Comment (vishalinfyom) [inline tests/Repositories/TaskRepositoryTest.php]: we are writing test cases for this kind of stuff, here some random project id is passed and still test is passing, this should not happen, so put validations that will check given project is exist or not

File: tests/Repositories/TaskRepositoryTest.php [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+        $getTask = $this->taskRepo->getTaskList();
+
+        $this->assertCount(2, $getTask);
+        $this->assertContains($tasks[0]->title, $getTask);
+        $this->assertContains($tasks[1]->title, $getTask);
+    }
+
+    /** @test */
+    public function it_can_retrieve_task_list_of_logged_in_user()
```
Comment (vishalinfyom) [inline tests/Repositories/TaskRepositoryTest.php]: `it_can_retrieve_task_list_of_given_projects`

File: tests/Repositories/TaskRepositoryTest.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+        $this->assertCount(2, $getTask->taskAssignee);
+        $this->assertEquals($prepareTask['assignees'][0], $getTask->taskAssignee[0]->id);
+        $this->assertEquals($prepareTask['assignees'][1], $getTask->taskAssignee[1]->id);
+    }
+
+    /** @test */
+    public function it_can_retrieve_task_list()
+    {
+        $project = factory(Project::class)->create();
```
Comment (vishalinfyom) [inline tests/Repositories/TaskRepositoryTest.php]: not need to create this

File: tests/Repositories/TaskRepositoryTest.php [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        ]);
+
+        $getTask = $this->taskRepo->getTaskList([$project->id => $project->name]);
+
+        $this->assertCount(2, $getTask);
+        $tasks->map(function (Task $task) use ($getTask) {
+            $this->assertContains($task->title, $getTask);
+        });
+
+        $this->assertArrayHasKey($project->id, $getTask);
```
Comment (vishalinfyom) [inline tests/Repositories/TaskRepositoryTest.php]: remove this

File: tests/Repositories/TaskRepositoryTest.php [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+        $tasks->map(function (Task $task) use ($getTask) {
+            $this->assertContains($task->title, $getTask);
+        });
+    }
+
+    /** @test */
+    public function it_can_retrieve_task_list_of_given_projects()
+    {
+        $project = factory(Project::class)->create();
```
Comment (vishalinfyom) [inline tests/Repositories/TaskRepositoryTest.php]: create one more task

File: tests/Repositories/TaskRepositoryTest.php [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+        $this->assertEquals($task['assignees'][1], $getTask->taskAssignee[1]->id);
+    }
+
+    /** @test */
+    public function test_can_update_task_with_tags_and_assignees()
+    {
+        $task = factory(Task::class)->create();
+
+        $prepareTask = $this->generateTaskInputs(['title' => 'random string']);
+
```
Comment (vishalinfyom) [inline tests/Repositories/TaskRepositoryTest.php]: remove extra blank lines

File: tests/Repositories/TaskRepositoryTest.php [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        $createdTask = $this->taskRepo->store($task);
+
+        $getTask = Task::with(['tags', 'taskAssignee'])->findOrFail($createdTask->id);
+
+        $this->assertEquals($task['title'], $getTask->title);
+        $this->assertEquals($task['tags'][0], $getTask->tags[0]->id);
+
+        $this->assertCount(2, $getTask->taskAssignee);
+        $this->assertEquals($task['assignees'][0], $getTask->taskAssignee[0]->id);
```
Comment (vishalinfyom) [inline tests/Repositories/TaskRepositoryTest.php]: better to use map here and below tests

File: tests/Repositories/TaskRepositoryTest.php [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+    {
+        // task with different project
+        factory(Task::class)->create();
+
+        $project = factory(Project::class)->create();
+        $tasks = factory(Task::class)->times(2)->create([
+            'project_id' => $project->id,
+        ]);
+
+        $getTask = $this->taskRepo->getTaskList([$project->id => $project->name]);
```
Comment (mitulgolakiya) [inline tests/Repositories/TaskRepositoryTest.php]: I do not understand this.

**Pull request merged**